### PR TITLE
deps!: use fixed cosmwasm version (v0.51.0)

### PR DESCRIPTION
**Pull request by MSalopek**
`wasmd` v0.50.x seems to have a regression that breaks amino signing on execute messages. The issue affects cosmos hub testnets running `gaiad v19.0.0-rc0` - mainnet is unaffected.

It seems that `wamd v0.50.x` is a broken release that did not get retracted so it can still be imported.
Nothing in the release notes suggested that it was a broken.

Context:
* https://github.com/CosmWasm/wasmd/pull/1844
* https://github.com/CosmWasm/wasmd/issues/1829

### CODE REVIEW

File: tests/integration/test_utils.go [hunk 1]
```diff
@@ -27,6 +28,28 @@ import (
 
 var app *gaiaApp.GaiaApp
 
+// Some tests require a random directory to be created when runnin IBC testing suite with gaia.
+// This is due to how CosmWasmVM initializes the VM - all IBC testing apps must have different dirs so they don't conflict.
+func GaiaAppIniterRandomDir() (ibctesting.TestingApp, map[string]json.RawMessage) {
```
Comment (p-offtermatt) [inline tests/integration/test_utils.go]: Maybe 
```suggestion
func GaiaAppIniterTempDir() (ibctesting.TestingApp, map[string]json.RawMessage) {
```
instead? The Randomness seems not the important part, it's that it's a dedicated directory

**Review by p-offtermatt — APPROVED** [review]
Tiny nit. LGTM otherwise

**Review by dusan-maksimovic — APPROVED** [review]

File: tests/integration/test_utils.go (hunk 1 above)
Comment (MSalopek) [reply tests/integration/test_utils.go]: Applied!

File: app/keepers/keepers.go [hunk 2]
```diff
@@ -469,6 +468,7 @@ func NewAppKeeper(
 	// Must be called on PFMRouter AFTER TransferKeeper initialized
 	appKeepers.PFMRouterKeeper.SetTransferKeeper(appKeepers.TransferKeeper)
 
+	wasmDir := filepath.Join(homePath, "wasm")
```
Comment (reecepbcups) [inline app/keepers/keepers.go:471]: :(

ref: same issue as I did on Juno: https://github.com/CosmosContracts/juno/pull/961

**Comment by reecepbcups** [comment]
New CI test needs to be written to ensure this does not happen again:

Write a CI which uploads, and executes/queries a contract. 
Then upgrade, then try to execute / query again on the SAME contract

This can not be caught if you upload a new contract, since it saved to the new (incorrect) directory. Which is easy to overlook

**Pull request merged**